### PR TITLE
runc-shim: support shim cgroup and oom score

### PR DESCRIPTION
**Pull request by Burning1020**
Signed-off-by: Zhang Tianyang <burning9699@gmail.com>

### CODE REVIEW

File: crates/runc-shim/src/service.rs [hunk 1]
```diff
@@ -81,7 +88,33 @@ impl Shim for Service {
             None => {}
         }
 
-        let address = spawn(opts, &grouping, Vec::new())?;
+        let (child_id, address) = spawn(opts, &grouping, Vec::new())?;
+
+        #[cfg(target_os = "linux")]
```
Comment (jiangliu) [inline crates/runc-shim/src/service.rs:93]: How about split out this as a function?

File: crates/runc-shim/src/service.rs [hunk 2]
```diff
@@ -144,6 +177,34 @@ impl Shim for Service {
     }
 }
 
+/// Sets the OOM score for the process to the parents OOM score + 1
```
Comment (jiangliu) [inline crates/runc-shim/src/service.rs:180]: All these functions may be grouped as a submodule, so we could reduce conditional compilation.

File: crates/runc-shim/src/service.rs (hunk 1 above)
Comment (Burning1020) [reply crates/runc-shim/src/service.rs:93]: I was planning to do that, but worried about changing too much. Now that you have the same feeling , I'd like to try again :)

File: crates/runc-shim/src/service.rs (hunk 2 above)
Comment (Burning1020) [reply crates/runc-shim/src/service.rs:180]: You're right. Should I mark `cgroup` as a new custom feature to replace `target_os = "linux"` at the same time?

**Comment by Burning1020** [comment]
updated. ping @jiangliu

**Review by mxpv — APPROVED** [review]

**Pull request merged**